omod/src/main/java/org/openmrs/module/referenceapplication/page/controller/LoginPageController.java

```diff
@@ -253,22 +291,21 @@ public String post(@RequestParam(value = "username", required = false) String us
 
 					return "redirect:" + ui.pageLink(ReferenceApplicationConstants.MODULE_ID, "home");
 				}
-			}
-			catch (ContextAuthenticationException ex) {
-				if (log.isDebugEnabled())
-					log.debug("Failed to authenticate user");
-
+			} else if (sessionLocation == null) {
 				pageRequest.getSession().setAttribute(ReferenceApplicationWebConstants.SESSION_ATTRIBUTE_ERROR_MESSAGE,
-				    ui.message(ReferenceApplicationConstants.MODULE_ID + ".error.login.fail"));
+						ui.message("referenceapplication.login.error.locationRequired"));
+			} else {
+				// the UI shouldn't allow this, but protect against it just in case
+				pageRequest.getSession().setAttribute(ReferenceApplicationWebConstants.SESSION_ATTRIBUTE_ERROR_MESSAGE,
+						ui.message("referenceapplication.login.error.invalidLocation", sessionLocation.getName()));
 			}
+		}
+		catch (ContextAuthenticationException ex) {
+			if (log.isDebugEnabled())
```

Comment:
Did you take a look at this? https://wiki.openmrs.org/display/docs/Java+Conventions#JavaConventions-CodeStyle

Reply:
Actually, this came from the existing code. Do you want me to remove this as well?